airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/report_streams/report_streams.py

```diff
@@ -385,39 +386,41 @@ def _init_reports(self, profile: Profile, report_date: str) -> List[ReportInfo]:
-            # different metric list for each record.
-            request_record_type = record_type.split("_")[0]
-            self.logger.info(f"Initiating report generation for {profile.profileId} profile with {record_type} type for {report_date} date")
-            response = self._send_http_request(
-                urljoin(self._url, self.report_init_endpoint(request_record_type)),
-                profile.profileId,
-                report_init_body,
-            )
-            if response.status_code != self.report_is_created:
-                error_msg = f"Unexpected HTTP status code {response.status_code} when registering {record_type}, {type(self).__name__} for {profile.profileId} profile: {response.text}"
-                if self._skip_known_errors(response):
-                    self.logger.warning(error_msg)
-                    break
-                raise ReportInitFailure(error_msg)
-
-            response = ReportInitResponse.parse_raw(response.text)
-            report_infos.append(
-                ReportInfo(
-                    report_id=response.reportId,
-                    record_type=record_type,
-                    profile_id=profile.profileId,
-                    status=Status.IN_PROGRESS,
-                    metric_objects=[],
+            for report_init_body in self._get_init_report_body(report_date, record_type, profile):
+                if not report_init_body:
+                    continue
+                # Some of the record types has subtypes. For example asins type
+                # for  product report have keyword and targets subtypes and it
+                # represented as asins_keywords and asins_targets types. Those
+                # subtypes have mutually excluded parameters so we requesting
+                # different metric list for each record.
+                request_record_type = record_type.split("_")[0]
+                self.logger.info(
+                    f"Initiating report generation for {profile.profileId} profile with {record_type} type for {report_date} date"
                 )
-            )
-            self.logger.info("Initiated successfully")
+                response = self._send_http_request(
+                    urljoin(self._url, self.report_init_endpoint(request_record_type)),
+                    profile.profileId,
+                    report_init_body,
+                )
+                if response.status_code != self.report_is_created:
+                    error_msg = f"Unexpected HTTP status code {response.status_code} when registering {record_type}, {type(self).__name__} for {profile.profileId} profile: {response.text}"
+                    if self._skip_known_errors(response):
+                        self.logger.warning(error_msg)
+                        break
+                    raise ReportInitFailure(error_msg)
+
+                response = ReportInitResponse.parse_raw(response.text)
+                report_infos.append(
```

Comment:
can we re-name `report_infos` > `report_info` ?

Reply:
renamed `report_infos`  to `report_info_list`